File: services/graphql-tms/src/models/vehicle_maintenance.rs
use super::sea_orm_active_enums::VehicleServiceTypeEnum;

#[derive(Clone, Debug, PartialEq, Eq, async_graphql :: SimpleObject)]
pub struct Model {
    pub id: Uuid,
    pub vehicle_id: Uuid,
    pub service_date: Date,
    pub service_type: Option<VehicleServiceTypeEnum>,
    pub cost: Option<Decimal>,
    pub notes: Option<String>,
    pub created_at: Option<DateTime>,
    pub updated_at: Option<DateTime>,
}

File: packages/db/src/schemas/tms/vehicle_maintenance.ts
import { z } from 'zod'
import { TmsVehicleServiceTypeEnum } from '../../db.types'

export const VehicleMaintenanceSchema = z.object({
  id: z.uuid({ message: 'Invalid UUID format for ID' }),
  vehicleId: z.uuid({ message: 'Invalid UUID format for vehicle ID' }),
  cost: z.coerce
    .number({ message: 'Cost must be a number' })
    .min(0, { error: 'Cost must be at least 0' })
    .max(1000000, { error: 'Cost must be at most 1,000,000' }),
  notes: z.string({ message: 'Notes must be a string' }).optional().nullable(),
  serviceDate: z.date({ message: 'Invalid date format for service date' }),
  serviceType: z
    .enum(TmsVehicleServiceTypeEnum, {
      message: 'Invalid vehicle service type',
    })
    .optional()
    .nullable(),
  createdAt: z.date({ message: 'Invalid date format for created at' }).optional().nullable(),
  updatedAt: z.date({ message: 'Invalid date format for updated at' }).optional().nullable(),
})

export type TmsVehicleMaintenance = z.infer<typeof VehicleMaintenanceSchema>

export const VehicleMaintenanceInsertSchema = VehicleMaintenanceSchema.omit({
  id: true,
  createdAt: true,
  updatedAt: true,
})

export const VehicleMaintenanceUpdateSchema = VehicleMaintenanceInsertSchema.partial()
